examples/rvc-app/rvc-common/include/rvc-device.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "rvc-mode-delegates.h"
+#include "rvc-operational-state-delegate.h"
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class RvcDevice
+{
+private:
+    RvcRunMode::RvcRunModeDelegate mRunModeDelegate;
+    ModeBase::Instance mRunModeInstance;
+
+    RvcCleanMode::RvcCleanModeDelegate mCleanModeDelegate;
+    ModeBase::Instance mCleanModeInstance;
+
+    RvcOperationalState::RvcOperationalStateDelegate mOperationalStateDelegate;
+    OperationalState::Instance mOperationalStateInstance;
+
+public:
+    RvcDevice() :
+        mRunModeDelegate(),
+        mRunModeInstance(&mRunModeDelegate, 0x1, RvcRunMode::Id, chip::to_underlying(RvcRunMode::Feature::kOnOff)),
+        mCleanModeDelegate(),
+        mCleanModeInstance(&mCleanModeDelegate, 0x1, RvcCleanMode::Id, chip::to_underlying(RvcCleanMode::Feature::kOnOff)),
+        mOperationalStateDelegate(), mOperationalStateInstance(&mOperationalStateDelegate, 0x01, RvcOperationalState::Id)
+    {
+        // set start-up modes and state
```

Comment:
It looks like the lines below only set the current state, not the startup mode. Shouldn't this happen according to the StartUpMode setting?

Reply:
The text might be a bit confusing given that there is an attribute called start-up-mode. I don't mean that this is going to set the start-up-mode attribute, I meant that this sets the current-mode at start up. This example doesn't enable the start-up-mode attribute. We always want the same behaviour and so there is no need to implement the start-up-mode attribute. I will update the text.

Reply:
See [here](https://github.com/project-chip/connectedhomeip/pull/28719/commits/63c23e20c8f4e9df44048c7839ec41c0d41280f9).

Reply:
This will be added in the next iteration of this app which will be coming soon.